Stub definition of __cxa_thread_atexit_impl

alastairreid opened the pull request:
This function is usually a weak symbol but, in KLEE, this behaves like a call
to an unknown function and chaos ensues.
Worse, it happens just as the program is cleanly shutting itself down,
so programs that are cleanly exiting crash with the wrong message.

ccadar commented:
Thanks, @alastairreid .  Do you have a small test case for this?  
And just wondering, shouldn't this function be defined in libcxx?

alastairreid commented:
oops, forgot to add a testcase - will add that and push another update.

Good question about where to put it - I had not thought much about it.
I think that adding it to the existing runtime/klee-libc/__cxa_atexit.c might be the right place.

ccadar commented:
Thanks again, @alastairreid , let's add this. But can you move the test to a new `test/Runtime/KLEE-Libc` directory? (as it's unrelated to uclibc).

alastairreid commented:
Rebased and squashed.
(I'm still trying to get used to the way github-based code review works. Rebasing and squashing fits my preferred style better.)

MartinNowack reviewed (approved):
@alastairreid Excellent! Thanks for this PR. Going to merge.
